src/js/utils/request.js

```diff
@@ -204,9 +207,148 @@ const request = {
     query.outFields = outFields;
     query.outStatistics = [stat];
     return task.execute(query);
-  }
+  },
+
+  getRecentTiles(params) {
+    const deferred = new Deferred();
+
+    if (!params.start || !params.end) {
+      // If no date, use the default.
+      params.start = moment().subtract(3, 'months').format('YYYY-MM-DD');
+      params.end = moment().format('YYYY-MM-DD');
+    }
+
+    const recentTilesUrl = new URL(urls.satelliteImageService);
+    Object.keys(params).forEach(key => recentTilesUrl.searchParams.append(key, params[key]));
+
+    const fetchTiles = (count = 0) => {
+      return fetch(
+        recentTilesUrl,
+        {
+          method: 'GET',
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+        }
+      ).then(res => res.json())
+       .then(res => {
+         return new Promise((resolve) => {
+          setTimeout(() => {
+            if (res.errors && res.errors[0].status !== 200 && count < 25) {
+              count++;
+              resolve(fetchTiles(count));
+            }
+            resolve(res);
+          }, 100);
+        });
+      });
+    };
 
+    fetchTiles().then(response => {
+      if (response.errors) {
+        deferred.reject(response);
+        return;
+      }
+      deferred.resolve(response);
+    });
+    return deferred;
+  },
+
+  getImageryData(params, tiles) {
+    const deferred = new Deferred();
+    const sourceData = [];
+    tiles.forEach((tile) => {
+      sourceData.push({ source: tile.attributes.source });
+    });
 
+    // Create request body that will be used for both the recent-tiles/tiles
+    // request and the recent-tiles/thumbs request
+    const content = {
+      bands: params.bands,
+      source_data: sourceData,
+    };
+
+    const postTiles = (count = 0) => {
```

Comment:
@jenboyd Can we hoist `postTiles` and `postThumbs` up to the same independent level as their parent functions so that they are not defined inside of a `request` helper function?